DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,189 @@
+/*
+ *
+* This is a part of CTPPS offline software.
+* Author:
+*   Fabrizio Ferro (ferro@ge.infn.it)
+*   Enrico Robutti (robutti@ge.infn.it)
+*   Fabio Ravera   (fabio.ravera@cern.ch)
+*
+*/
+
+#ifndef DataFormats_CTPPSReco_CTPPSPixelLocalTrack_H
+#define DataFormats_CTPPSReco_CTPPSPixelLocalTrack_H
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelRecHit.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
+
+
+#include "TVector3.h"
+#include "TMatrixD.h"
+#include "TVectorD.h"
+
+//----------------------------------------------------------------------------------------------------
+
+class CTPPSPixelLocalTrack
+{
+  public:
+    class CTPPSPixelFittedRecHit: public CTPPSPixelRecHit
+    {
+      public:
+        CTPPSPixelFittedRecHit(const CTPPSPixelRecHit &hit, const TVector3 &space_point_on_det, std::pair<double,double> residual, std::pair<double,double> pull) :
+            CTPPSPixelRecHit(hit), space_point_on_det_(space_point_on_det), residual_(residual), pull_(pull), isUsedForFit_(false), isRealHit_(false) {}
+    
+        CTPPSPixelFittedRecHit() : CTPPSPixelRecHit(), residual_(std::pair<double,double>(0,0)), pull_(std::pair<double,double>(0,0)), isUsedForFit_(false), isRealHit_(false) {}
+    
+        virtual ~CTPPSPixelFittedRecHit() {}
+    
+        inline const TVector3 & getGlobalCoordinates() const { return space_point_on_det_; }
+        inline void setGlobalCoordinates(const TVector3 & space_point_on_det) { space_point_on_det_ = space_point_on_det; }
+    
+        inline double getXResidual() const { return residual_.first; }
+        inline double getYResidual() const { return residual_.second; }
```

Comment:
It may be more transparent to access to x and y by using LocalPoint type for them instead of the pair

Reply:
Done